h2/src/main/org/h2/jdbc/JdbcPreparedStatement.java

```diff
@@ -485,6 +489,15 @@ public void setObject(int parameterIndex, Object x, int targetSqlType)
             if (isDebugEnabled()) {
                 debugCode("setObject("+parameterIndex+", x, "+targetSqlType+");");
             }
+            if (targetSqlType == Types.OTHER) {
+                if (x instanceof UUID) {
```

Comment:
Probably here check for isPostgreSQL() is needed as well? Otherwise looks like a breaking change. We need a test for a table with column type OTHER, insert UUID there and try to do some operations with this value.